pkg/services/ngalert/migration/alert_rule.go

```diff
@@ -9,29 +9,27 @@ import (
 	"github.com/prometheus/common/model"
 
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+
 	"github.com/grafana/grafana/pkg/infra/log"
 	legacymodels "github.com/grafana/grafana/pkg/services/alerting/models"
+	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/services/datasources"
-	migmodels "github.com/grafana/grafana/pkg/services/ngalert/migration/models"
 	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/store"
 	"github.com/grafana/grafana/pkg/tsdb/graphite"
 	"github.com/grafana/grafana/pkg/util"
 )
 
-func addLabelsAndAnnotations(l log.Logger, alert *legacymodels.Alert, dashboardUID string, channels []*legacymodels.AlertNotification) (data.Labels, data.Labels) {
+func addLabelsAndAnnotations(l log.Logger, alert *legacymodels.Alert, dashboardUID string) (data.Labels, data.Labels) {
 	tags := alert.GetTagsFromSettings()
-	lbls := make(data.Labels, len(tags)+len(channels)+1)
+	lbls := make(data.Labels, len(tags)+1)
 
 	for _, t := range tags {
 		lbls[t.Key] = t.Value
 	}
 
 	// Add a label for routing
 	lbls[ngmodels.MigratedUseLegacyChannelsLabel] = "true"
-	for _, c := range channels {
-		lbls[contactLabel(c.Name)] = "true"
-	}
```

Comment:
It looks like the MigratedUseLegacyChannelsLabel above is also related to channels. I think it makes sense to move it too.

Reply:
Adding the `MigratedUseLegacyChannelsLabel` label is not conditional on anything, it always needs to be there so that default legacy notification channels (using `alertname~=.+`) will match. Given that, I slightly lean toward leaving it here but don't feel strongly about it.